src/smt/theory_str_noodler/theory_str_noodler.cpp

```diff
@@ -887,7 +901,7 @@ namespace smt::noodler {
 
         // it is possible that the arithmetic formula becomes unsatisfiable already by adding the (underapproximating)
         // length constraints from initial assignment
-        expr_ref lengths = len_node_to_z3_formula(dec_proc.get_initial_lengths());
+        lengths = len_node_to_z3_formula(dec_proc.get_initial_lengths());
         if(check_len_sat(lengths) == l_false) {
             STRACE("str", tout << "Unsat from initial lengths" << std::endl);
             block_curr_len(lengths, true, true);
```

Comment:
Shouldn't this be removed now?

Reply:
Well, I don't know. I think there might be cases when you get something that is unsatisfiable after preprocessing steps. I suggest to keep it there (or try if there is some practical effect).